### PR TITLE
use DT_INST_PROP over DT_INST_PROP_OR if possible

### PR DESCRIPTION
**Pull request by justpresident**
This closes following issue: https://github.com/zephyrproject-rtos/zephyr/issues/67651

It might happens that DT(_INST)_PROP_OR is used with boolean properties. For instance:

	.single_wire = DT_INST_PROP_OR(index, single_wire, false),	\
	.tx_rx_swap = DT_INST_PROP_OR(index, tx_rx_swap, false),	\

This is not required as boolean properties are generated with false value when not present, so the _OR macro extension is superflous and the above code can be replaced by:

	.single_wire = DT_INST_PROP(index, single_wire),		\
	.tx_rx_swap = DT_INST_PROP(index, tx_rx_swap),			\

The changes were generated with regexps:
```
find . -type f -name *.c | xargs -n1 perl -pi -e 's/DT_PROP_OR\((.+?),\s*\\?\s*(.+?),\s*\\?\s*false\)/DT_PROP($1, $2)/m'
find . -type f -name *.c | xargs -n1 perl -pi -e 's/DT_INST_PROP_OR\((.+?),\s*\\?\s*(.+?),\s*\\?\s*false\)/DT_INST_PROP($1, $2)/m'
```

### CODE REVIEW

**Review by aaronemassey — DISMISSED** [review]
TIL

**Review by ifyall — DISMISSED** [review]

**Review by ycsin — COMMENTED** [review]
and now the `\` are not aligned anymore :(

**Comment by justpresident** [comment]
@aaronemassey , could you please help me to figure out what is the problem with tests. I attached a build.log from ```west twister -p qemu_x86 -T tests/drivers/fuel_gauge```
[build.log](https://github.com/zephyrproject-rtos/zephyr/files/14037568/build.log)

File: drivers/fuel_gauge/sbs_gauge/emul_sbs_gauge.c [hunk 1]
```diff
@@ -367,7 +367,7 @@ static int emul_sbs_sbs_gauge_init(const struct emul *target, const struct devic
 	static struct sbs_gauge_emul_data sbs_gauge_emul_data_##n;                                 \
 	static const struct sbs_gauge_emul_cfg sbs_gauge_emul_cfg_##n = {                          \
 		.addr = DT_INST_REG_ADDR(n),                                                       \
-		.cutoff_support = DT_PROP_OR(DT_DRV_INST(n), battery_cutoff_support, false),       \
+		.cutoff_support = DT_PROP(DT_DRV_INST(n), battery_cutoff_support),       \
```
Comment (ycsin) [inline drivers/fuel_gauge/sbs_gauge/emul_sbs_gauge.c:370]: the binding doesnt have `battery_cutoff_support` prop

**Review by erwango — DISMISSED** [review]
It would be great to review the `\` now broken indentation as mentioned by @ycsin. Otherwise LGTM

**Review by ycsin — CHANGES_REQUESTED** [review]
please try to align those `\` affected by this PR, it will be harder to get people to care if more of them become unaligned

**Review by erwango — DISMISSED** [review]
Thanks for this update!

**Comment by justpresident** [comment]
Sorry for the trouble with multiple updates. Next time I will submit a draft diff first

**Review by ycsin — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
(unassigning and merging, does not belong to any particular area anyway)

**Comment by gmarull** [comment]
This change unpapered bugs in the updated driver, which does not compile after the changes in this PR https://github.com/zephyrproject-rtos/zephyr/pull/68300. Twister may need some AI on how to choose the right tests to run.

File: drivers/fuel_gauge/sbs_gauge/emul_sbs_gauge.c (hunk 1 above)
Comment (aaronemassey) [reply drivers/fuel_gauge/sbs_gauge/emul_sbs_gauge.c]: Sorry was sick last week. Looking today.

**Pull request merged**